### PR TITLE
Cleanup MinecraftAPI.cs

### PR DESCRIPTION
**Pull request by OoLunar**
Cleans up MinecraftAPI.cs by using the Json extension methods. Additionally marks some return types as nullable

### CODE REVIEW

**Review by Naamloos — APPROVED** [review]
LGTM, don't see anything weird.

**Comment by OoLunar** [comment]
The httpClient methods used could optionally be modified to allow a CancellationToken, however that'd require modifying the method parameters and would thus be a breaking change

**Review by Tides — CHANGES_REQUESTED** [review]
The json extension methods should take in our `Globals.JsonOptions` for the serializer options param (To avoid any de/serialization problems)

**Review by Seb-stian — APPROVED** [review]
I suppose we should make those string literals `static readonly`, but that may be beyond the scope of this PR. I like the changes.

File: Obsidian/Utilities/Mojang/MinecraftAPI.cs [hunk 1]
```diff
@@ -1,49 +1,43 @@
...
 using System.Text;
-using System.Text.Json;
 using System.Web;
 
 namespace Obsidian.Utilities.Mojang;
 
 public class MinecraftAPI
 {
-    private static readonly HttpClient Http = Globals.HttpClient;
+    private static readonly HttpClient httpClient = new();
```
Comment (Craftplacer) [inline Obsidian/Utilities/Mojang/MinecraftAPI.cs]: Why is a new instance made inside `MinecraftAPI`?

Comment (OoLunar) [reply Obsidian/Utilities/Mojang/MinecraftAPI.cs]: Missed that. Meant to use the Global HttpClient when adding the Global Json Options

Comment (OoLunar) [reply Obsidian/Utilities/Mojang/MinecraftAPI.cs]: Fixed as of commit 63905f2

**Review by Craftplacer — APPROVED** [review]

**Pull request merged**